### PR TITLE
Ros fb conversion test

### PR DESCRIPTION
**Pull request by jarkenau**

### CODE REVIEW

File: seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+
+    // TransformStamped Start
+    geometry_msgs::TransformStamped original_t_stamped = createTransformStamped();
+
+    flatbuffers::grpc::Message<seerep::fb::TransformStamped> fb_transform_stamped;
+    fb_transform_stamped = seerep_ros_conversions_fb::toFlat(original_t_stamped, p_uuid);
+
+    geometry_msgs::TransformStamped converted_t_stamped =
+        seerep_ros_conversions_fb::toROS(*fb_transform_stamped.GetRoot());
+    // Transform End
```
Comment (mhoellmann) [inline seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp:406]: break this down into multiple methods.

File: seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+geometry_msgs::Vector3Stamped rosToFbConversionTest::original_v3_stamped;
+geometry_msgs::Vector3Stamped rosToFbConversionTest::converted_v3_stamped;
+
+// Transform
+geometry_msgs::Transform rosToFbConversionTest::original_t;
+geometry_msgs::Transform rosToFbConversionTest::converted_t;
+
+// TransformStamped
+geometry_msgs::TransformStamped rosToFbConversionTest::original_t_stamped;
+geometry_msgs::TransformStamped rosToFbConversionTest::converted_t_stamped;
```
Comment (mhoellmann) [inline seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp:458]: Is this really necessary? Isn't it easier to create an object of the class?
Would the structure of the test benefit from putting everything in the class?

**Review by mhoellmann — COMMENTED** [review]
please rebase to the master branch or cherry-pick the relevant commits. This PR contains a lot of commits from the previous PR.

File: seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp [hunk 3]
```diff
@@ -405,57 +432,43 @@ class rosToFbConversionTest : public testing::Test
         seerep_ros_conversions_fb::toROS(*fb_transform_stamped.GetRoot());
     // Transform End
   }
-};
 
-// Objects created below are used by the TEST functions
+  /**
+   * @brief This function instantiates all the attributes and fills them with arbitrary values.
+   * */
+  void TestBody()
```
Comment (mhoellmann) [inline seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp:439]: Do not use this function according to the gTest documentation. Can you not better call those functions in the constructor?

>   // Runs the test after the test fixture has been set up.
  //
  // A sub-class must implement this to define the test logic.
  //
  // DO NOT OVERRIDE THIS FUNCTION DIRECTLY IN A USER PROGRAM.
  // Instead, use the TEST or TEST_F macro.
  virtual void TestBody() = 0;

File: seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp [hunk 4]
```diff
@@ -477,7 +488,7 @@ void testHeader(std_msgs::Header original_header, std_msgs::Header converted_hea
 TEST_F(rosToFbConversionTest, testHeader)
 {
   // expect that original and converted-from-fb are equal
-  testHeader(original_header, converted_header);
+  testHeader(original_header, testClass.converted_header);
```
Comment (mhoellmann) [inline seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp:491]: Why do you need the "testClass." here but at no other converted object?

File: seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp (hunk 3 above)
Comment (mhoellmann) [reply seerep-ros/seerep_ros_conversions_fb/test/ros_to_fb_conversion_test.cpp:439]: Maybe SetUpTestSuite() is the correct methode?

**Review by mhoellmann — CHANGES_REQUESTED** [review]

**Review by mhoellmann — APPROVED** [review]

**Comment by 104H** [comment]
closes #185 and #82

**Pull request merged**